### PR TITLE
Remove old elasticsearch libs

### PR DESCRIPTION
**Pull request by esoergel**
@emord remove the last instance of [`es_fakes`](https://github.com/dimagi/commcare-hq/pull/9707), so I deleted it, and did some more clean-up of old elasticsearch tools, moving them to `ESQuery`.

`es_wrapper` was something I wrote when I first started using elasticsearch, but it was eventually superceded by `ESQuery`.  There were just a few places remaining where it was in-use, so I switched those over and deleted it.

It'd be nice to unify the rest of the places that use elasticsearch too, but I don't think there's any particular urgency to it.

commit-by-commit will make sense, but whatever floats your boat.
@emord

### CODE REVIEW

File: corehq/elastic.py [hunk 1]
```diff
@@ -134,9 +134,9 @@ def run_query(url, q):
     return get_es().get(url, data=q)
 
 
-def es_histogram(histo_type, domains=None, startdate=None, enddate=None, tz_diff=None,
-        interval="day", q=None, filters=[]):
-    q = q or {"query": {"match_all":{}}}
+def es_histogram(histo_type, domains=None, startdate=None, enddate=None,
+        interval="day", filters=[]):
+    q = {"query": {"match_all":{}}}
```
Comment (dimagimon) [inline corehq/elastic.py]: E231 missing whitespace after ':'

File: corehq/apps/api/resources/v0_5.py [hunk 2]
```diff
@@ -28,6 +27,16 @@
...
 
 
+def user_es_call(domain, q, fields, size, start_at):
+    query = (UserES()
+             .domain(domain)
+             .set_query({"query_string": {"query": q}})
+             .fields(fields)
+             .size(size)
+             .start(start_at))
+    return query.run().hits
```
Comment (esoergel) [inline corehq/apps/api/resources/v0_5.py]: I just wrote a small function to translate the `es_wrapper` call to `ESQuery`, as there's already a test mock that uses that API.

File: corehq/elastic.py [hunk 3]
```diff
@@ -134,9 +134,9 @@ def run_query(url, q):
     return get_es().get(url, data=q)
 
 
-def es_histogram(histo_type, domains=None, startdate=None, enddate=None, tz_diff=None,
-        interval="day", q=None, filters=[]):
-    q = q or {"query": {"match_all":{}}}
+def es_histogram(histo_type, domains=None, startdate=None, enddate=None,
+        interval="day", filters=[]):
```
Comment (esoergel) [inline corehq/elastic.py]: `es_histogram` is only used in one place, and that call doesn't use the `tz_diff` or `q` parameters, so I removed them.  It'd be nice to switch this function over, but I'm less familiar with that and just decided to punt for now.

File: corehq/elastic.py [hunk 4]
```diff
@@ -245,88 +242,6 @@ def convert(param):
     return result
 
 
-def es_wrapper(index, domain=None, q=None, doc_type=None, fields=None,
```
Comment (czue) [inline corehq/elastic.py]: :skull:

**Comment by czue** [comment]
> It'd be nice to unify the rest of the places that use elasticsearch too, but I don't think there's any particular urgency to it.

fyi @esoergel I'm also looking into whether we should consider switching over to [Elasticsearch DSL](https://elasticsearch-dsl.readthedocs.org/en/latest/). It looks like the goal is to provide a lot of ORM-like functionality on top of ES (so similar to what `ESQuery` does). if you do end up diving into optimizing ES querying it might make sense to look at it, or at least please keep me and @dimagi/scale-team in the loop

**Comment by czue** [comment]
(we're blocked on the 1.0 migration for now so this would be down the road)

**Comment by esoergel** [comment]
@czue Sounds good.  I don't have any plans to do more with this right now, I just saw the opportunity to remove this one with relatively little effort, so I went for it.

Lemme know if you end up liking Elasticsearch DSL.  A couple questions I have offhand are:
* Looks like it uses kwargs for terms filters - how does it handle nested lookups?  Ie, you can't do `q.filtered("term", form.meta.userID="1234")`, does it map `__` to `.` or something?
* Can we extend it?  I really think we should write things like filters once and abstract them, to avoid mistakes like messing up "domain" and "domain.exact" (and "domain_memberships.domain.exact"), and because it's easier to write queries if you have trusted components to build from.

**Comment by czue** [comment]
i don't know the answers to those things offhand but will note them down for whenever we end up looking closer at it. assume we could extend it, though have no idea whether it's designed to allow that.

**Comment by dannyroberts** [comment]
Late to the :tada: but :+1: :heart_eyes: the :goberserk:

**Pull request merged**